### ISSUE TITLE
lein-ancient doesn't handle composite profiles

### ISSUE THREAD

**Issue by belucid** [xsc/lein-ancient#61]
Composite profiles ([described here](https://github.com/technomancy/leiningen/blob/master/doc/PROFILES.md#composite-profiles)) are not handled by lein-ancient. Any profile that is a composite has its dependencies and plugins skipped.

I'm happy to take the stab at fixing this, but I wanted you to possibly weigh in on how you wanted it fixed since we can go in 2 directions.

1) One possible fix is to have the reader drop the vector out, since it's not interesting to lein-ancient, and thereby make composite profiles look the same as any other profiles.

2) Another possible fix is to leave reader alone, and have the composite profile come in with the interstitial vector, and update another part(s) of lein ancient to deal with a profile that isn't a map with `:plugins` and `:dependencies` but is instead a vector with said map as the 2nd item in the vector. I haven't looked into the code enough to know where this would need to happen.

I wrote a new (failing) test case for reader of what option 1 would look like [here](https://github.com/belucid/lein-ancient/commit/45b115bb7583f3efbe4d57ffe074a3a15e94de8f).

Seems the simplest change would be option 1. If you want me to go that route, I'll adjust reader so that the test passes.

If you'd prefer I go with option 2, that's fine, just let me know.

**Comment by xsc** [comment]
@belucid True, that doesn't get handled well. I think the fix belongs somewhere into [this abomination of a function](https://github.com/xsc/lein-ancient/blob/e26ed97e014286ae1ee2c0e3df2651618fddf4da/src/leiningen/ancient/artifact/check.clj#L23-L56) (which is very much in need of refactoring).

It takes a map of `{:dependencies ..., :plugins ..., :profiles ...}` and spits out a seq of `{:path ..., :artifact ...}` - and `:path` is e.g.:

```clojure
[:profiles :dev :dependencies 0]
````

This has to be adapted to allow a path into a composite profile, e.g.:

```clojure
[:profiles :dev 1 :dependencies 0]
```

If you want to tackle that (and thus this issue), just let me know - otherwise, I'll surely get around to it.

**Comment by belucid** [comment]
That is a lovely function :smile: !

I'll take a crack at it and let you know where I get with it.

**Comment by xsc** [comment]
Could you try out the latest SNAPSHOT?

```clojure
[lein-ancient "0.6.8-SNAPSHOT"]
```

**Comment by belucid** [comment]
I tried the snapshot. Given a project file like this (the numbers are labels so we can talk about each spot):

```clojure
{
  :dependencies [1]
  :plugins [2]

  :profiles {
    :dev {
      :dependencies [3]
      :plugins [4]
    }

    :prof1 {
      :dependencies [5]
      :plugins [6]
    }
  
    :prof2 [:prof1 {
      :dependencies [7]
      :plugins [8]
    }
  }
}
```
I created out of date dependencies and plugins in each spot and out of date plugins in each spot were found. That's great!

Here are the results:

0.6.7
lein ancient :allow-qualified - finds 1, 3, 5
lein ancient :allow-qualified :plugins - finds 2, 4, 6

0.6.8-SNAPSHOT
lein ancient :allow-qualified - finds 1, 3, 5, 7
lein ancient :allow-qualified :plugins - finds 2, 4, 6, 8

:+1: 

Thanks for that. Good work.
